main_script.py

```diff
@@ -0,0 +1,140 @@
+from abc import ABC, abstractmethod
+from Bio import SeqIO, SeqUtils 
+import os
+
+
+class ThisIsNotTheRightSequence(ValueError):
+    pass
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__(self):
+        pass
+    
+    @abstractmethod
+    def __getitem__(self, index):
+        pass
+    
+    @abstractmethod
+    def get_slice(self, start, end):
+        pass
+    
+    @abstractmethod
+    def __str__(self):
+        pass
+    
+    @abstractmethod
+    def check_the_alphabet(self):
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, sequence, gc_content=None):
+        self.sequence = sequence
+        self.gc_content = gc_content
+
+    def __len__(self):
+        return len(self.sequence)
+    
+    def __getitem__(self, index):
+        return self.sequence[index]
+    
+    def get_slice(self, start, end):
+        return self.sequence[start:end]
+    
+    def __str__(self):
+        return self.sequence
+
+    def check_the_alphabet(self):
+        if set(self.sequence).issubset(type(self).alphabet.keys()):
+            return True
+        else:
+            raise ThisIsNotTheRightSequence()
+    
+    def count_gc_content(self):
+        gc_count = 0
+        for i in self.sequence:
+            if i == 'g' or i == 'c' or i == 'G' or i == 'C':
+                gc_count += 1
+        self.gc_content = gc_count * 100 / len(self.sequence)
+        return self.gc_content
+
+    def complement(self):
+        complement_seq = ''
+        for i in self.sequence:
+            complement_seq += type(self).alphabet.get(i)
```

Comment:
```suggestion
            complement_seq += type(self).alphabet[i]
```
Все таки в 99% случаев к словарям обращаются через квадратные скобки. Как минимум они не промолчат, если что то пойдет не так